### PR TITLE
Leon get performances

### PR DESCRIPTION
**Pull request by rogers89**
* Refactoring firebase -> http client
* Incorporated HTTP tablet logic
* Next step: Adjudication forms

### CODE REVIEW

File: app/src/main/java/com/uwblueprint/dancefest/PerformanceActivity.kt [hunk 1]
```diff
@@ -122,91 +100,60 @@ class PerformanceActivity : AppCompatActivity() {
         return true
     }
 
-    private fun fetchData(querySnapshot: QuerySnapshot) {
+    private fun fetchData() {
```
Comment (imruahmed) [inline app/src/main/java/com/uwblueprint/dancefest/PerformanceActivity.kt]: Can we rename this to be more specific?

File: app/src/main/java/com/uwblueprint/dancefest/PerformanceActivity.kt [hunk 2]
```diff
@@ -55,65 +46,52 @@ class PerformanceActivity : AppCompatActivity() {
...
-                            Log.e(TAG, "Failed to listen to performance collection.", error)
-                            return@addSnapshotListener
-                        }
-                        if (querySnapshot == null) {
-                            Log.e(TAG, "Null querySnapshot")
-                        } else {
-                            fetchData(querySnapshot)
-                        }
+        // Initialize tablet id here
+        dancefestClientAPI.call(api.getOrCreateTablet(Build.SERIAL)) {
```
Comment (imruahmed) [inline app/src/main/java/com/uwblueprint/dancefest/PerformanceActivity.kt:72]: Can we move this into its own function for readability

File: app/src/main/java/com/uwblueprint/dancefest/PerformanceActivity.kt [hunk 3]
```diff
@@ -55,65 +46,52 @@ class PerformanceActivity : AppCompatActivity() {
...
+                            object : TabLayout.OnTabSelectedListener {
+                                override fun onTabReselected(tab: TabLayout.Tab) {}
+                                override fun onTabUnselected(tab: TabLayout.Tab) {}
+                                override fun onTabSelected(tab: TabLayout.Tab) {
+                                    view_pager.currentItem = tab.position
+                                }
+                            })
                     }
+
+                    fetchData()
```
Comment (imruahmed) [inline app/src/main/java/com/uwblueprint/dancefest/PerformanceActivity.kt]: Do we need this here or is it taken care of in the `onResume` ??

**Review by imruahmed — APPROVED** [review]
left some nits, otherwise looks good

File: app/src/main/java/com/uwblueprint/dancefest/PerformanceActivity.kt (hunk 3 above)
Comment (rogers89) [reply app/src/main/java/com/uwblueprint/dancefest/PerformanceActivity.kt]: seems `onResume` is called after `onCreate`, good eye

**Pull request merged**